host_core/lib/host_core/providers/provider_module.ex

```diff
@@ -113,6 +114,18 @@ defmodule HostCore.Providers.ProviderModule do
     {:noreply, state}
   end
 
+  def handle_info({:DOWN, _ref, :port, _port, :normal}, state) do
+    Logger.debug("Received DOWN message from port (executable stopped normally)")
+
+    {:noreply, state}
+  end
+
+  def handle_info({:DOWN, _ref, :port, _port, reason}, state) do
+    Logger.error("Received DOWN message from port (executable stopped) - #{reason}")
+    publish_provider_stopped(state.public_key, state.link_name, state.instance_id, "#{reason}")
+    {:stop, reason, state}
+  end
```

Comment:
If a provider fails to start in its startup routine (e.g. if we start the provider that fails to deserialize the base64 encoded HostData) would that be handled by this handler? If so, that's great.

Reply:
yeah, if the provider crashes reading stdin, this will be triggered.